test/optim/test_optimize.py

```diff
@@ -1562,6 +1597,22 @@ def test_optimize_acqf_mixed_empty_ff(self):
                 raw_samples=10,
             )
 
+    def test_optimize_acqf_mixed_return_best_only_q2(self):
+        with self.assertRaises(
+            NotImplementedError,
+            msg="`return_best_only=False` is only supported for q=1.",
+        ):
+            mock_acq_function = MockAcquisitionFunction()
```

Comment:
nit: isolate the expected Exception in the context manager
```suggestion
        mock_acq_function = MockAcquisitionFunction()
        with self.assertRaises(
            NotImplementedError,
            msg="`return_best_only=False` is only supported for q=1.",
        ):
```